filebeat/harvester/reader/multiline.go

```diff
@@ -215,12 +216,18 @@ func (mlr *Multiline) resetBuffer(m Message) {
 	mlr.addLine(m)
 	// Timestamp of first message is taken as overall timestamp
 	mlr.ts = m.Ts
-	mlr.fields = m.Fields
+	if m.Fields != nil {
+		mlr.fields = m.Fields
+	} else {
+		mlr.fields = common.MapStr{}
+	}
 }
 
 // finalize writes the existing content into the returned message and resets all reader variables.
 func (mlr *Multiline) finalize() Message {
 
+	mlr.fields["lines"] = mlr.numLines
```

Comment:
is numLines correctly updated? What if max lines has been reached? Predicate in `addLine` reads really weird.